molgenis-api-data/src/main/java/org/molgenis/api/data/v3/DataServiceV3.java

```diff
@@ -60,4 +63,25 @@
    * @throws UnknownEntityException if no entity exists for the given entity identifier
    */
   void delete(String entityTypeId, String entityId);
+
+  /**
+   * Delete one entity.
+   *
+   * @param entityTypeId entity type identifier
+   * @param query query specifing which
+   * @throws UnknownRepositoryException if no repository exists for the given entity type identifier
+   * @throws UnknownEntityException if no entity exists for the given entity identifier
+   */
+  void delete(String entityTypeId, Query query);
+
+  List<Entity> find(
```

Comment:
How about removing ```int count(String entityTypeId, Query query);``` and let ```find``` return a EntityCollection with List<Entity> and int total?